src/python/pants/source/source_root.py

```diff
@@ -18,6 +19,7 @@ class SourceRootCategories:
 
 
 SourceRoot = namedtuple('_SourceRoot', ['path', 'langs', 'category'])
+SourceRootWithLangDirnames = namedtuple('SourceRootWithLangDirnames', ['path', 'langs', 'category'])
```

Comment:
This name is possibly more confusing than useful. How about `UncanonicalizedSourceRoot` and add a comment explaining what that is.